File: frontend/lib/dto/card_dto/UpdateSecurityOptionRequest.dart
class UpdateSecurityOptionRequest {
  final int cardId;
  final bool? contactlessEnabled;
  final bool? ecommerceEnabled;
  final bool? tpeEnabled;

  UpdateSecurityOptionRequest({
    required this.cardId,
    this.contactlessEnabled,
    this.ecommerceEnabled,
    this.tpeEnabled,
  });

  Map<String, dynamic> toJson() {
    return {
      'cardId': cardId,
      if (contactlessEnabled != null) 'contactlessEnabled': contactlessEnabled,
      if (ecommerceEnabled != null) 'ecommerceEnabled': ecommerceEnabled,
      if (tpeEnabled != null) 'tpeEnabled': tpeEnabled,
    };
  }
}

File: microservices/microservices/card-service/src/main/java/hps/ma/dto/UpdateSecurityOptionRequest.java
package hps.ma.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UpdateSecurityOptionRequest {
    private Long cardId;
    private Boolean contactlessEnabled;
    private Boolean ecommerceEnabled;
    private Boolean tpeEnabled;


}